File: framework/languages/java/samples/java/SupportChat/Server/Configuration/src/main/java/systems/zlink/samples/supportchat/server/configuration/SampleFlowLog.java
package systems.zlink.samples.supportchat.server.configuration;

public final class SampleFlowLog {
    private SampleFlowLog() {
    }

    public static String path(SampleTopology topology, String role) {
        return topology.requiredLogDirectory() + "/flow-" + role + ".log";
    }
}

File: framework/languages/java/samples/kotlin/DeliveryDispatch/Server/Configuration/src/main/kotlin/systems/zlink/samples/kotlin/deliverydispatch/server/configuration/SampleFlowLog.kt
package systems.zlink.samples.kotlin.deliverydispatch.server.configuration

object SampleFlowLog {
    fun roleReady(role: String, port: Int) {
        println("deliverydispatch role-ready role=$role port=$port")
    }

    fun deliveryAccepted(deliveryId: String, customerId: String) {
        println("deliverydispatch dispatch: accepted delivery=$deliveryId customer=$customerId")
    }

    fun statusRecorded(deliveryId: String, status: String, courierId: String) {
        println("deliverydispatch tracking: status delivery=$deliveryId status=$status courier=$courierId")
    }

    fun customerSubscribed(deliveryId: String) {
        println("deliverydispatch customer-session: subscribed delivery=$deliveryId")
    }
}
